### PR TITLE
Dev 120 manage sku selector options

### PR DESCRIPTION
**Pull request by AliveMen**
Disable unavailable options when selecting a product.
SKU selector

### CODE REVIEW

File: assets/js/product.js [hunk 1]
```diff
@@ -129,15 +135,67 @@ storefrontApp.controller('productController', ['$rootScope', '$scope', '$window'
...
+                var retVal = propMap1.hasOwnProperty(x);
+                if (retVal) {
+                    retVal = propMap1[x][0].value == master[x][0].value;
+                }
+                return retVal;
+            });
+        }
+
+
+        function filterVariationBySelectedProps(variations, selectedPropMap) {
```
Comment (t13ka) [inline assets/js/product.js]: 'filterVariationByProperties' is better than 'filterVariationBySelectedProps'

Comment (t13ka) [reply assets/js/product.js]: maybe 'getVariationByPropertyMaps' is better than 'filterVariationByProperties' )

File: assets/js/product.js [hunk 2]
```diff
@@ -129,15 +135,67 @@ storefrontApp.controller('productController', ['$rootScope', '$scope', '$window'
...
+                return partyComparePropertyMaps(getVariationPropertyMap(x), selectedPropMap);
+            });
+        }
+
+        function isPropertyAvailable(property, key) {
+            var allVariationsMaps = angular.copy($scope.allVariationPropsMap);
+            _.each(allVariationsMaps[key], function (prop) {
+                prop.selected = false;
+            });
+            var workProperty = _.find(allVariationsMaps[key], function (y) { return y.value === property.value; });
```
Comment (t13ka) [inline assets/js/product.js]: y ?

File: assets/js/product.js [hunk 3]
```diff
@@ -129,15 +135,67 @@ storefrontApp.controller('productController', ['$rootScope', '$scope', '$window'
...
+            
+        }
+
         //Method called from View when user clicks one property value
         $scope.checkProperty = function (property) {
             //Select appropriate property and unselect previous selection
+            
+           // var propertyDisplayName = property.displayName;
             _.each($scope.allVariationPropsMap[property.displayName], function (x) {
                 x.selected = x != property ? false : !x.selected;
```
Comment (t13ka) [inline assets/js/product.js]: x?

File: assets/js/product.js [hunk 4]
```diff
@@ -129,15 +135,67 @@ storefrontApp.controller('productController', ['$rootScope', '$scope', '$window'
...
+            _.each(_.keys($scope.allVariationPropsMap), function (key) {
+                _.each($scope.allVariationPropsMap[key], function (prop) {
+
+                     if($scope.allVariationPropsMap[key].length == 1) {
+                         $scope.allVariationPropsMap[key][0].selected = true;
+                    }
+                    else {
+                        var available = isPropertyAvailable(prop, key);
+                        var originalProperty = _.find($scope.allVariationPropsMap[key], function (y) { return y.value === prop.value; });
+                        originalProperty.available = available;
```
Comment (t13ka) [inline assets/js/product.js]: Can we use this form? This is more readable:
originalProperty.available = isPropertyAvailable(prop, key);

File: assets/js/product.js [hunk 5]
```diff
@@ -129,15 +135,67 @@ storefrontApp.controller('productController', ['$rootScope', '$scope', '$window'
...
+                    }
+                });
+           });
+
             //try to find the best variation match for selected properties
             $scope.selectedVariation = findVariationBySelectedProps(allVariations, getSelectedPropsMap($scope.allVariationPropsMap));
+            if ($scope.selectedVariation) {
+                console.log('selected variation', $scope.selectedVariation);
```
Comment (t13ka) [inline assets/js/product.js]: For what we should log this information in a production environment?

File: assets/js/product.js [hunk 6]
```diff
@@ -129,15 +135,67 @@ storefrontApp.controller('productController', ['$rootScope', '$scope', '$window'
...
+            }
+            return false;
+            
+        }
+
         //Method called from View when user clicks one property value
         $scope.checkProperty = function (property) {
             //Select appropriate property and unselect previous selection
+            
+           // var propertyDisplayName = property.displayName;
```
Comment (t13ka) [inline assets/js/product.js]: if this code should be in the comment then why we leave this?

File: assets/js/product.js [hunk 7]
```diff
@@ -129,15 +135,67 @@ storefrontApp.controller('productController', ['$rootScope', '$scope', '$window'
             });
         }
 
+        function partyComparePropertyMaps(propMap1, master) {
+            return _.every(_.keys(master), function (x) {
+                var retVal = propMap1.hasOwnProperty(x);
```
[the remark this answers is not in the document]
Comment (AliveMen) [reply assets/js/product.js]: removed method partyComparePropertyMaps

File: assets/js/product.js (hunk 3 above)
Comment (AliveMen) [reply assets/js/product.js]: renamed

File: assets/js/product.js (hunk 5 above)
Comment (AliveMen) [reply assets/js/product.js]: removed

File: assets/js/product.js (hunk 6 above)
Comment (AliveMen) [reply assets/js/product.js]: removed

File: assets/js/product.js (hunk 2 above)
Comment (AliveMen) [reply assets/js/product.js]: renamed

File: assets/js/product.js (hunk 1 above)
Comment (AliveMen) [reply assets/js/product.js]: renamed

**Pull request merged**